ui/react-components/components/AppointmentEditor/AppointmentEditor.jsx

```diff
@@ -103,21 +157,55 @@ const AppointmentEditor = props => {
         const startTimeBeforeEndTime = isStartTimeBeforeEndTime(startTime, endTime);
         setPatientError(!isValidPatient);
         setServiceError(!service);
-        setDateError(!startDate);
+        setAppointmentDateError(!appointmentDate);
         setStartTimeError(!startTime);
         setEndTimeError(!endTime);
         setStartTimeBeforeEndTimeError(!startTimeBeforeEndTime);
-        return isValidPatient && service && startDate && startTime && endTime && startTimeBeforeEndTime;
+        return isValidPatient && service && appointmentDate && startTime && endTime && startTimeBeforeEndTime;
     };
 
     const {angularState} = React.useContext(AppContext);
 
+    const isValidRecurringAppointment = () => {
```

Comment:
Can we extract validations into seperate files like validations

Reply:
yes, we have that thought. Will take it